From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Tom Levy <tomlevy93@gmail.com>
Date: Fri, 24 Jul 2020 09:13:22 +1200
Subject: Revert "Create files with correct ownership instead of using 'chown
 -R'"

It will be re-done slightly differently soon.

This reverts commit c1a9224dbdc96c0146758fdfb017ed8ad67f43fe.
---
 docker/domserver/Dockerfile | 26 ++++++++++----------------
 docker/domserver/build.sh   |  2 +-
 2 files changed, 11 insertions(+), 17 deletions(-)

diff --git a/docker/domserver/Dockerfile b/docker/domserver/Dockerfile
index 9f7420e65..9814706b9 100644
--- a/docker/domserver/Dockerfile
+++ b/docker/domserver/Dockerfile
@@ -31,19 +31,17 @@ ADD https://getcomposer.org/installer composer-setup.php
 RUN php composer-setup.php \
     && mv /composer.phar /usr/local/bin/composer
 
-RUN mkdir /domjudge-src
-
 # Install PHP dependencies
 # (also done by 'make dist', but we do it here *before* adding the rest of the source to leverage the build cache)
-COPY --chown=domjudge composer.json composer.lock /domjudge-src/domjudge/
-RUN cd /domjudge-src/domjudge && sudo -u domjudge composer install --prefer-dist -o --no-scripts
+COPY composer.json composer.lock /domjudge-src/domjudge/
+RUN cd /domjudge-src/domjudge && composer install --prefer-dist -o --no-scripts
 
 # Add DOMjudge source code and build script
-COPY --chown=domjudge . /domjudge-src/domjudge
+COPY . /domjudge-src/domjudge
 ADD docker/domserver/build.sh /domjudge-src
 
 # Bootstrap the configure/build system
-RUN sudo -u domjudge make -C /domjudge-src/domjudge dist
+RUN make -C /domjudge-src/domjudge dist
 
 # Build and install domserver
 RUN /domjudge-src/build.sh
@@ -72,24 +70,20 @@ RUN apt-get update \
 	enscript lpr \
 	&& rm -rf /var/lib/apt/lists/*
 
-# Set up user
-RUN useradd -m domjudge
-
 # Install composer
 ADD https://getcomposer.org/installer composer-setup.php
 RUN php composer-setup.php \
     && mv /composer.phar /usr/local/bin/composer
 
-RUN mkdir /opt/domjudge && chown domjudge: /opt/domjudge
-
 # Copy domserver directory and add script files
-COPY --from=domserver-build --chown=domjudge /opt/domjudge/domserver /opt/domjudge/domserver
-COPY --from=domserver-build --chown=domjudge /opt/domjudge/doc /opt/domjudge/doc
+COPY --from=domserver-build /opt/domjudge/domserver /opt/domjudge/domserver
+COPY --from=domserver-build /opt/domjudge/doc /opt/domjudge/doc
 COPY ["docker/domserver/scripts", "/scripts/"]
 
-# Create PHP FPM socket dir, change ownership of some domjudge directories and fix scripts
-# ("chown -R domjudge: /opt/domjudge" not required; verify using "chown --changes ...")
-RUN mkdir -p /run/php \
+# Add user, create PHP FPM socket dir, change permissions for domjudge directory and fix scripts
+RUN useradd -m domjudge \
+	&& mkdir -p /run/php \
+	&& chown -R domjudge: /opt/domjudge \
 	&& chown -R www-data: /opt/domjudge/domserver/tmp \
 	# for DOMjudge <= 7.2 (submitdir was removed in commit DOMjudge/domjudge@d66725038)
 	&& if [ -d /opt/domjudge/domserver/submissions ]; then chown -R www-data: /opt/domjudge/domserver/submissions; fi \
diff --git a/docker/domserver/build.sh b/docker/domserver/build.sh
index 82270a551..f616c9e7b 100755
--- a/docker/domserver/build.sh
+++ b/docker/domserver/build.sh
@@ -1,7 +1,7 @@
 #!/bin/bash -e
 
 cd /domjudge-src/domjudge*
-# ("chown -R domjudge: ." not required; verify using "chown --changes ...")
+chown -R domjudge: .
 sudo -u domjudge ./configure -with-baseurl=http://localhost/
 
 # Passwords should not be included in the built image. We create empty files here to prevent passwords from being generated.
-- 
2.20.1

